### DISCUSSION TITLE
Allow to change the friction curves at runtime

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#98] [Ideas]
This is a feature request to allow changing the lateral/longitudinal friction curves at runtime. 

In arcade games, it can be useful to have an ability to change the friction mode while driving. A vehicle could have a tight traction with the ground for normal controls, but a game would allow "to enter drift mode" by hitting a handbreak, for example. This would adjust/change the curve with more lose friction values and revert back to normal as soon as vehicle gains full traction.

**Comment by jrouwe** [comment]
If you change the curves in WheelSettingsWV after construction the vehicle should immediately update (the wheels keep a reference to the original settings you used to create the wheels and you can get them back with ```wheel.GetSettings()```).

**Reply by LeXXik** [reply]
Right, thanks to your earlier hint, I realized I can cast `Wheel` to `WheelWV`, which allows then to access the curves.

**Comment by DennisSmolek** [comment]
Did you end up ever implementing this in your setup? I was about to play with making an api for my library to do just this...

**Reply by LeXXik** [reply]
Yep, I did. I allow the vehicle to change the friction at runtime, but it is very game specific. I think in most cases having a single model is sufficient.

**Reply by DennisSmolek** [reply]
How are you handling friction on different surfaces? Just setting them on the surface directly?

I was thinking of putting a texture on a plane or heightmap and sampling it to get a friction value. That way I could create like dirt on the edge of a track, grass etc without having to actually create new meshes for each material. 

Figured I'd have curves for each type and then just switch at runtime.. 

It's all theoretical/back burnered for a bit but on my mind.

**Reply by jrouwe** [reply]
I'm not sure if you need to modify those curves at all if this is all about surface friction, you can also use:

```
VehicleConstraintCallbacksJS.GetCombinedFriction(unsigned long inWheelIndex, ETireFrictionDirection inTireFrictionDirection, float inTireFriction, [Const, Ref] Body inBody2, [Const, Ref] SubShapeID inSubShapeID2);
```

to directly override the friction from a callback.
